test/install_test.go

```diff
@@ -57,7 +57,7 @@ var (
 
 		// k8s hitting readiness endpoints before components are ready
 		`.* linkerd-(controller|identity|grafana|prometheus|proxy-injector|web)-.*-.* linkerd-proxy ERR! \[ *\d+.\d+s\] proxy={server=in listen=0\.0\.0\.0:4143 remote=.*} linkerd2_proxy::(proxy::http::router service|app::errors unexpected) error: an error occurred trying to connect: Connection refused \(os error 111\) \(address: 127\.0\.0\.1:.*\)`,
-		`.* linkerd-(controller|identity|grafana|prometheus|proxy-injector|web)-.*-.* linkerd-proxy ERR! \[ *\d+.\d+s\] proxy={server=out listen=127\.0\.0\.1:4140 remote=.*} linkerd2_proxy::(proxy::http::router service|app::errors unexpected) error: an error occurred trying to connect: Connection refused \(os error 111\) \(address: .*:4191\)`,
+		`.* linkerd-(controller|identity|grafana|prometheus|proxy-injector|web)-.*-.* linkerd-proxy ERR! \[ *\d+.\d+s\] proxy={server=out listen=127\.0\.0\.1:4140 remote=.*} linkerd2_proxy::(proxy::http::router service|app::errors unexpected) error: an error occurred trying to connect: Connection refused \(os error 111\) \(address: .*\)`,
```

Comment:
Just to make sure, we want to match on any IP address not just the port?

Reply:
Correct, we get arbitrary IPs here, have a look at:
https://travis-ci.org/linkerd/linkerd2/jobs/514816958#L7642-L7671

```
    install_test.go:359: Found error in prometheus/linkerd-proxy log: linkerd-git-a21c7be9 linkerd-prometheus-868996b5c9-n7bhl linkerd-proxy ERR! [     8.097256s] proxy={server=out listen=127.0.0.1:4140 remote=10.60.1.44:41090} linkerd2_proxy::app::errors unexpected error: an error occurred trying to connect: Connection refused (os error 111) (address: 10.60.5.10:9994)
=== RUN   TestRestarts
--- PASS: TestRestarts (0.22s)
FAIL
FAIL	command-line-arguments	92.726s
Running test [install_test.go] --proxy-auto-inject
=== RUN   TestVersionPreInstall
--- PASS: TestVersionPreInstall (0.20s)
=== RUN   TestCheckPreInstall
--- PASS: TestCheckPreInstall (0.68s)
=== RUN   TestInstall
--- PASS: TestInstall (40.80s)
=== RUN   TestVersionPostInstall
--- PASS: TestVersionPostInstall (0.21s)
=== RUN   TestInstallSP
--- PASS: TestInstallSP (0.90s)
=== RUN   TestCheckPostInstall
--- PASS: TestCheckPostInstall (1.09s)
=== RUN   TestDashboard
--- PASS: TestDashboard (1.47s)
=== RUN   TestInject
--- PASS: TestInject (16.51s)
=== RUN   TestServiceProfileDeploy
--- PASS: TestServiceProfileDeploy (0.69s)
=== RUN   TestCheckProxy
--- PASS: TestCheckProxy (1.36s)
=== RUN   TestLogs
--- FAIL: TestLogs (30.02s)
    install_test.go:359: Found error in prometheus/linkerd-proxy log: linkerd-git-a21c7be9-auto-inject linkerd-prometheus-7dbb594dbf-nw8rk linkerd-proxy ERR! [     4.844423s] proxy={server=out listen=127.0.0.1:4140 remote=10.60.5.11:38260} linkerd2_proxy::app::errors unexpected error: an error occurred trying to connect: Connection refused (os error 111) (address: 10.60.4.176:9996)
    install_test.go:359: Found error in prometheus/linkerd-proxy log: linkerd-git-a21c7be9-auto-inject linkerd-prometheus-7dbb594dbf-nw8rk linkerd-proxy ERR! [     6.582321s] proxy={server=out listen=127.0.0.1:4140 remote=10.60.5.11:57700} linkerd2_proxy::app::errors unexpected error: an error occurred trying to connect: Connection refused (os error 111) (address: 10.60.4.177:3000)
```

Reply:
Honestly, it's probably pretty fragile to have specific OS error codes in here as well.